src/components/forms/TextInput/TextInput.test.tsx

```diff
@@ -11,6 +11,47 @@ describe('TextInput component', () => {
     expect(queryByTestId('textInput')).toBeInTheDocument()
   })
 
+  describe('masking features', () => {
+    const setup = () => {
+      const utils = render(
+        <TextInput
+          id="input-type-alphanumeric"
+          name="input-type-alphanumeric"
+          type="text"
+          aria-describedby="hint-alphanumeric"
+          mask="___ ___"
+          pattern="\w\d\w \d\w\d"
+          charset="A#A #A#"
+        />
+      )
+      const input = screen.getByTestId('textInput')
+      const mask = screen.getByTestId('input-type-alphanumericMask')
+      return {
+        input,
+        mask,
+        ...utils,
+      }
+    }
+
+    it('renders with class and placeholder', () => {
+      const { input, mask } = setup()
+      expect(input).toHaveClass('usa-masked')
+      expect(mask).toHaveTextContent('___ ___')
+    })
+
+    it('autoformats inputted text', () => {
+      const { input } = setup()
+      fireEvent.change(input, { target: { value: 'A1B2C3' } })
```

Comment:
**issue (convention, test-robustness)**: `userEvent` is preferred over `fireEvent` which would looks something like this (you'll need to `async` the test function and add the import from `@testing-library/user-event`)

more info: https://testing-library.com/docs/user-event/intro/

> fireEvent dispatches DOM events, whereas user-event simulates full interactions, which may fire multiple events and do additional checks along the way.

```suggestion
      const user = userEvent.setup()
      await user.type(input, 'A1B2C3')
```